test/test_blocks.py

```diff
@@ -531,6 +533,15 @@ def generic_update():
             }
             assert output["data"][1] == {"__type__": "update", "mode": "dynamic"}
 
+    def test_error_raised_if_num_outputs_mismatch(self):
+        with gr.Blocks() as demo:
+            textbox1 = gr.Textbox()
+            textbox2 = gr.Textbox()
+            button = gr.Button()
+            button.click(lambda x: x, textbox1, [textbox1, textbox2])
+        with pytest.raises(ValueError):
```

Comment:
nit: I would `match=` against the expected error message here but maybe I'm just too paranoid 😂